hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/utils/FederationStateStoreFacade.java

```diff
@@ -1088,4 +1089,21 @@ public ApplicationSubmissionContext getApplicationSubmissionContext(ApplicationI
   public FederationCache getFederationCache() {
     return federationCache;
   }
+
+  /**
+   * Delete the mapping of home {@code SubClusterId} of a previously submitted
+   * {@code ApplicationId}. Currently response is empty if the operation was
+   * successful, if not an exception reporting reason for a failure.
+   *
+   * @param appId ApplicationId.
+   */
+  public void deleteApplicationHomeSubCluster(ApplicationId appId) {
+    try {
+      DeleteApplicationHomeSubClusterRequest request =
+          DeleteApplicationHomeSubClusterRequest.newInstance(appId);
+      stateStore.deleteApplicationHomeSubCluster(request);
+    } catch (Exception e) {
+      LOG.error("deleteApplicationHomeSubCluster error, applicationId = {}.", appId, e);
```

Comment:
Shouldn't we surface this somehow?

Reply:
Thank you for your suggestion! I will handle exceptions in the code calling this function.